lib/Agrammon/Outputs/FilterGroupCollection.pm6

```diff
@@ -54,30 +57,57 @@ class Agrammon::Outputs::FilterGroupCollection {
         self.Numeric.Real
     }
 
-    #| Get a list of pairs mapping filter groups into the total value for that group.
-    method results-by-filter-group() {
-        [%!values-by-filter.map({ .key.filters => .value })]
+    #| Get a list of pairs mapping filter groups into the total value for that
+    #| group. If :all is passed, then all filters that could possibly have been
+    #| selected will be included, even if no instance used them, and they will
+    #| have a zero value.
+    method results-by-filter-group(Bool :$all = False) {
+        if $all {
+            if %!values-by-filter && !$!provenance {
+                die "Can only get all values when provenance of the filter group was provided";
+            }
+            my @results;
+            for $!provenance.keys.sort.reverse -> $filter-set {
```

Comment:
Unfortunately, this `sort` is not going to do anything useful. It is a `sort` on instances of `Agrammon::Model::FilterSet`, and will sort on memory location or some such. :-)

Reply:
I did 437fbcc18 but I don't know if it's what you actually want. :-)